tests/terratest/functions/set/setEKS.go

```diff
@@ -0,0 +1,105 @@
+	tfBlockBody := tfBlock.Body()
+	reqProvsBlock := tfBlockBody.AppendNewBlock("required_providers", nil)
+	reqProvsBlockBody := reqProvsBlock.Body()
+
+	reqProvsBlockBody.SetAttributeValue("rancher2", cty.ObjectVal(map[string]cty.Value{
+		"source":  cty.StringVal("rancher/rancher2"),
+		"version": cty.StringVal(terraformConfig.ProviderVersion),
+	}))
+	rootBody.AppendNewline()
+
+	// Provider
+	provBlock := rootBody.AppendNewBlock("provider", []string{"rancher2"})
+	provBlockBody := provBlock.Body()
+	provBlockBody.SetAttributeValue("api_url", cty.StringVal(`https://`+rancherConfig.Host))
+	provBlockBody.SetAttributeValue("token_key", cty.StringVal(rancherConfig.AdminToken))
+	provBlockBody.SetAttributeValue("insecure", cty.BoolVal(*rancherConfig.Insecure))
+	rootBody.AppendNewline()
+
+	// Resource cloud credential
+	cloudCredBlock := rootBody.AppendNewBlock("resource", []string{"rancher2_cloud_credential", "rancher2_cloud_credential"})
+	cloudCredBlockBody := cloudCredBlock.Body()
+	cloudCredBlockBody.SetAttributeValue("name", cty.StringVal(terraformConfig.CloudCredentialName))
+	ec2CredConfigBlock := cloudCredBlockBody.AppendNewBlock("amazonec2_credential_config", nil)
+	ec2CredConfigBlockBody := ec2CredConfigBlock.Body()
+	ec2CredConfigBlockBody.SetAttributeValue("access_key", cty.StringVal(terraformConfig.AWSAccessKey))
+	ec2CredConfigBlockBody.SetAttributeValue("secret_key", cty.StringVal(terraformConfig.AWSSecretKey))
+	ec2CredConfigBlockBody.SetAttributeValue("default_region", cty.StringVal(terraformConfig.Region))
+	rootBody.AppendNewline()
+
+	// Resource cluster
+	clusterBlock := rootBody.AppendNewBlock("resource", []string{"rancher2_cluster", "rancher2_cluster"})
+	clusterBlockBody := clusterBlock.Body()
+	clusterBlockBody.SetAttributeValue("name", cty.StringVal(terraformConfig.ClusterName))
+	eksConfigBlock := clusterBlockBody.AppendNewBlock("eks_config_v2", nil)
+	eksConfigBlockBody := eksConfigBlock.Body()
+	cloudCredID := hclwrite.Tokens{
+		{Type: hclsyntax.TokenIdent, Bytes: []byte(`rancher2_cloud_credential.rancher2_cloud_credential.id`)},
+	}
+	eksConfigBlockBody.SetAttributeRaw("cloud_credential_id", cloudCredID)
+	eksConfigBlockBody.SetAttributeValue("region", cty.StringVal(terraformConfig.Region))
+	eksConfigBlockBody.SetAttributeValue("kubernetes_version", cty.StringVal(k8sVersion))
+	awsSubnetsList := format.ListOfStrings(terraformConfig.AWSSubnets)
+	eksConfigBlockBody.SetAttributeRaw("subnets", awsSubnetsList)
+	awsSecGroupsList := format.ListOfStrings(terraformConfig.AWSSecurityGroups)
+	eksConfigBlockBody.SetAttributeRaw("security_groups", awsSecGroupsList)
+	eksConfigBlockBody.SetAttributeValue("private_access", cty.BoolVal(terraformConfig.PrivateAccess))
+	eksConfigBlockBody.SetAttributeValue("public_access", cty.BoolVal(terraformConfig.PublicAccess))
+
+	// Resource node groups
+	num := 1
+	for _, pool := range nodePools {
+		poolNum := strconv.Itoa(num)
+		if pool.DesiredSize <= 1 {
```

Comment:
[clarification]:
I thought aws / eks allowed size 1 pools? like, when I create through the UI, I am allowed to enter pool size = 1

Reply:
Let me follow-up w/ you on this.  With both Rancher and AWS Console, you can provision a pool size of 1.  With terraform rancher2 provider, you may not.  Not sure if this is by design, or a bug.  I will look into this and follow-up shortly.

Reply:
uhh.. so I can in fact spin up EKS w/ 1 node using rancher2 provider.  During development, this was not the case, and that validation was added to prevent users from encountering an error.  I will remove this validation and allow EKS to be provisioned w/ 1 node, as i'm not encountering any errors in doing so now.